rsmtool/input.py

```diff
@@ -860,15 +860,24 @@ def load_and_filter_data(csv_file,
     logger = logging.getLogger(__name__)
 
     # read the csv file into a data frame but we want to make
-    # sure to read in the `id_column`, candidate_column and
+    # sure to read in the `id_column`, `candidate_column` and
     # subgroups (if any) as a string to ensure
     # that we do not lose information, e.g., initial zeros
-
     string_columns = [id_column, candidate_column] + subgroups
     converter_dict = dict([(column, str) for column in string_columns if column])
 
+    # read in the CSV file
     df = pd.read_csv(csv_file, converters=converter_dict)
 
+    # it is possible for the `id_column` and `candidate_column` to be
+    # set to the same column name in the CSV file, e.g., if there is
+    # only one response per candidate. If this happens, we neeed to
+    # create a duplicate column for candidates for the downstream
+    # processing to work as usual.
+    if id_column == candidate_column:
+        df['candidate'] = df['ID'].copy()
```

Comment:
This should <<df[id_column]>> (the test pass because this is what you used as ID column)

I think we should have this bit (877-879) after checking that all specified names for all columns present in the data (that is between <<if missing columns:>> and <<df=rename...>>, else the tool will raise an obscure ValueError if the user gave wrong column name in both cases.